include/seastar/core/reactor.hh

```diff
@@ -282,6 +283,7 @@ private:
 
     boost::container::static_vector<std::unique_ptr<task_queue>, max_scheduling_groups()> _task_queues;
     internal::scheduling_group_specific_thread_local_data _scheduling_group_specific_data;
+    shared_mutex _scheduling_group_keys_mutex;
```

Comment:
This presumes that all calls to the scheduling group management functions happen on the same shard.

Reply:
No, the callers are shard-local already.